iModelCore/ecobjects/PublicApi/ECObjects/ECJsonUtilities.h

```diff
@@ -359,18 +372,35 @@ struct ECJsonUtilities
     //! Converts an id from a JSON value to a BeInt64Id
     //! @remarks The JSON must contain the Id value in one of the formats of BentleyApi::ECN::ECJsonInt64Format.
     //! @param[in] json JSON value containing the id
-    //! @return Resulting BeInt64Id. In case of error, an invalid BeInt64Id will be returned.
+    //! @return Resulting BeInt64Id. In case of error, an int64 < 0, a double/float that cannot be converted losslessly to uint64, or a json string with '-' or '.' an invalid BeInt64Id will be returned.
     template<class TBeInt64Id>
     static TBeInt64Id JsonToId(RapidJsonValueCR json)
         {
         int64_t val = 0;
-        if (SUCCESS != JsonToInt64(val, json))
+        bool stringCheckFailed = false;
+        if (json.IsString()) 
+            {
+            Utf8CP strVal = json.GetString();
+            if (strVal[0] == '-') // negative numbers are not valid
+                stringCheckFailed = true;
+            else if (strchr(strVal, '.') != nullptr) // decimal numbers are not valid
+                stringCheckFailed = true;
+            }
+        else if (json.IsFloat() && IsLosslessUint64(json.GetFloat())) 
+            {
+            return TBeInt64Id((uint64_t) json.GetFloat());
+            } 
+        else if (json.IsDouble() && IsLosslessUint64(json.GetDouble())) 
+            {
+            return TBeInt64Id((uint64_t) json.GetDouble());
+            }
+
+        if (stringCheckFailed || json.IsFloat() || json.IsDouble() || SUCCESS != JsonToInt64(val, json) || val < 0)
             {
             TBeInt64Id invalidId;
             invalidId.Invalidate();

```

Comment:
```suggestion

```

Nit and not your change, but default constructor produces an invalid Id already.